### PR TITLE
Prometheus: Update package version to v0.301.0

### LINKED ISSUE #98238: Prometheus: Update prometheus package version to v0.300.2
When it's released prometheus package version should be updated to v0.300.2 in order to support new utf8 metrics and labels syntax primarily with scopes feature.

### PR DESCRIPTION
**Pull request by itsmylife**
**What is this feature?**

It's updating the prometheus package version to the latest one `v0.300.2`.
The latest one has all the the features required to support utf8 metrics and labels. 

Currently the last version is `v0.300.1`. And because of the time of the year, the next release will be done in the new year. 
The new release will have this https://github.com/prometheus/prometheus/pull/15531

So then we will be able to parse the promql query on the backendside even it has utf8 metrics or labels.

**Why do we need this feature?**

To be able to parse utf8 labels and metrics especially with scopes feature

**Who is this feature for?**

Prometheus developers and scopes feature users.

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/98238

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -5,11 +5,6 @@ go 1.23.1
 // contains openapi encoder fixes. remove ASAP
 replace cuelang.org/go => github.com/grafana/cue v0.0.0-20230926092038-971951014e3f // @grafana/grafana-as-code
 
-// Override Prometheus version because Prometheus v2.X is tagged as v0.X for Go modules purposes and Go assumes
-// that v1.Y is higher than v0.X, so when we resolve dependencies if any dependency imports v1.Y we'd
-// import that instead of v0.X even though v0.X is newer.
-replace github.com/prometheus/prometheus => github.com/prometheus/prometheus v0.52.0
-
```
Comment (itsmylife) [inline go.mod:12]: I think we can use v0.300.x in everywhere and no need this replace anymore.

File: go.mod [hunk 2]
```diff
@@ -141,7 +136,7 @@ require (
 	github.com/prometheus/client_golang v1.20.5 // @grafana/alerting-backend
 	github.com/prometheus/client_model v0.6.1 // @grafana/grafana-backend-group
 	github.com/prometheus/common v0.61.0 // @grafana/alerting-backend
-	github.com/prometheus/prometheus v1.8.2-0.20221021121301-51a44e6657c3 // @grafana/alerting-backend
```
Comment (itsmylife) [inline go.mod:144]: I think I can update this to the latest version too.

File: go.work [hunk 3]
```diff
@@ -30,3 +30,5 @@ replace github.com/getkin/kin-openapi => github.com/getkin/kin-openapi v0.125.0
 replace github.com/prometheus/alertmanager => github.com/grafana/prometheus-alertmanager v0.25.1-0.20240930132144-b5e64e81e8d3
 
 replace github.com/crewjam/saml => github.com/grafana/saml v0.4.15-0.20240917091248-ae3bbdad8a56
+
+replace github.com/prometheus/prometheus => github.com/prometheus/prometheus v0.300.1
```
Comment (itsmylife) [inline go.work:34]: This will be `v0.300.2` when `v0.300.2`'s released

**Comment by NWRichmond** [comment]
@itsmylife is this ready to come out of draft mode?

**Comment by itsmylife** [comment]
> @itsmylife is this ready to come out of draft mode?

No not ready. Even I'll make it ready it will fail with the unit tests. We need to wait for prometheus `v0.300.2`.
I created the draft to get feedback on my removal of replace statement in go.mod file. 
Also as I updated prometheus version for all grafana, there are some errors here and there. I need to fix them too.

**Comment by itsmylife** [comment]
Waiting this PR to be resolved and merged first https://github.com/grafana/grafana/pull/96925

**Pull request closed without merging**